examples/dt/uni04delta-ipv6/control-plane/kustomization.yaml

```diff
@@ -4,5 +4,6 @@ kind: Kustomization
 components:
   - ../../../../dt/uni04delta-ipv6
 resources:
+  - coo-sub.yaml
```

Comment:
This works to inject the `Subscription` for COO, because it will get processed here [1].  I do wonder though if it would be cleaner to have a separate stage for it before the control-plane one.  It's not a deal-breaker in my opinion, however.  Maybe @fultonj and @hjensas have other thoughts.

[1] https://github.com/openstack-k8s-operators/architecture/blob/6c766ab2086ad407a55d5daab7bf602bdb187655/automation/vars/uni04delta-ipv6.yaml#L17-L28

Reply:
I think a separate step would be better.

Also, I think we should move the coo-sub.yaml to the library, since this is something that multiple VA/DTs will use? Something in `lib/olm-deps`, and add the kustomize to apply in `examples/common` ?

Then the first automation stage for VA/DT's with Telemetry would be to apply `path: examples/common/coo-sub`? (I think that should work, but we may need to relax the assertions in ci-framework's `kustomize_deploy`[1] role to allow automation stages without values?)


[1] https://github.com/openstack-k8s-operators/ci-framework/blob/main/roles/kustomize_deploy/tasks/execute_step.yml#L10-L11

Reply:
Those sound like good ideas to me

Reply:
I like the idea too. I'll see what I can do.

Reply:
Actually by moving the coo subscription to lib/olm-deps, it'll get picked up by examples/common/olm automatically. I assume that examples/common is used for each dt/va, so this way COO will be installed everywhere. I think that's OK to have it everywhere since it's listed as a dependency in the RHOSO docs and I've seen some efforts of enabling telemetry on all dts. Check it in the latest commit.

Reply:
I rechecked my testproject downsteam which depends on these changes. It's struggling to deploy the dataplane for quite some time now (I think the hypervisor is running out of resources), but it got far enough to successfully deploy the control plane and I can confirm that this approach for deploying COO worked. I'll reach out once I manage to get a successful run of the testproject.

Reply:
@hjensas are you ok with this most recent update?

Reply:
Yes, I think it is fine. We will end up install COO always, but I saw the resource usage was discussed and agreed to be small enough to do it like this.